python/runfiles/README.md

```diff
@@ -0,0 +1,90 @@
+# bazel-runfiles library
+
+This is a Bazel Runfiles lookup library for Bazel-built Python binaries and tests.
+
+Learn about runfiles: read [Runfiles guide](https://bazel.build/extending/rules#runfiles)
+or watch [Fabian's BazelCon talk](https://www.youtube.com/watch?v=5NbgUMH1OGo).
+
+## Importing
+
+The Runfiles API is available from two sources, a direct Bazel target, and a [pypi](https://pypi.org/) package.
+
+## Pure Bazel imports
+
+1. Depend on this runfiles library from your build rule, like you would other third-party libraries:
+
+    ```python
+    py_binary(
+        name = "my_binary",
+        # ...
+        deps = ["@rules_python//python/runfiles"],
+    )
+    ```
+
+2. Import the runfiles library:
+
+    ```python
+        from rules_python.python.runfiles import Runfiles
+    ```
+
+## Pypi imports
+
+1. Add the 'bazel-runfiles' dependency along with other third-party dependencies, for example in your `requirements.txt` file.
+
+2. Depend on this runfiles library from your build rule, like you would other third-party libraries:
+    ```python
+    load("@pip_deps//:requirements.bzl", "requirement")
+    
+    py_binary(
+        name = "my_binary",
+        ...
+        deps = [requirement("bazel-runfiles")],
+    )
+    ```
+
+3. Import the runfiles library:
+    ```python
+    from runfiles import Runfiles
+    ```
+
+## Typical Usage
+
+Create a `Runfiles` object and use `Rlocation` to look up runfile paths:
+
+```python
+r = Runfiles.Create()
+# ...
+with open(r.Rlocation("my_workspace/path/to/my/data.txt"), "r") as f:
+    contents = f.readlines()
+    # ...
+```
+
+The code above creates a manifest- or directory-based implementation based on the environment variables in `os.environ`. See `Runfiles.Create()` for more info.
+
+If you want to explicitly create a manifest- or directory-based
+implementation, you can do so as follows:
+
+```python
+r1 = Runfiles.CreateManifestBased("path/to/foo.runfiles_manifest")
+
+r2 = Runfiles.CreateDirectoryBased("path/to/foo.runfiles/")
+```
+
+If you want to start subprocesses, and the subprocess can't automatically
+find the correct runfiles directory, you can explicitly set the right
+environment variables for them:
+
+```python
+import subprocess
+from rules_python.python.runfiles import Runfiles
```

Comment:
FYI, within bzlmod this import isn't working. If we strip the rules_python segment it should work in both.

Reply:
Why doesn’t that work? That’s gonna be a pretty big change to make :(

Reply:
That is because the name of the directory that contains the library in bzlmod is not rules_python, but something else. Documenting that the import path should be `python.runfiles` for now should be good enough for now. 

If we want a specific import path to be `bazel_runfiles` instead or make `rules_python.python.runfiles` work in bzlmod, we should probably address this in a separate PR. Relying on the repository name in python imports tends to be brittle over time and standardising on a single import that is consistent with the PyPI library would be better, but I would argue that can be done as a separate PR - adding a folder bazel_runfiles under the python/runfiles dir and adding the necessary re-exports and aliases to it is fairly straightforward if that's the path we want nto go with.

Reply:
It looks like this is still the same? However the PR LGTM in general.

Reply:
I changed the import path in docs. If that’s not what you meant can you be more specific?

Reply:
I thought I still saw the import with rules_python in the subprocess example. I see that tho commit changed the docs elsewhere?

Reply:
```suggestion
from python.runfiles import Runfiles
```